scripts/prepdocs/files/listfilestrategy.py

```diff
@@ -0,0 +1,170 @@
+import base64
+import hashlib
+import os
+import re
+import tempfile
+from abc import ABC
+from glob import glob
+from typing import IO, AsyncGenerator, Dict, List, Optional, Union
+
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.storage.filedatalake.aio import (
+    DataLakeServiceClient,
+)
+
+
+class File:
+    """
+    Represents a file stored either locally or in a data lake storage account
+    This file might contain access control information about which users or groups can access it
+    """
+
+    def __init__(self, content: IO, acls: Optional[dict[str, list]] = None):
+        self.content = content
+        self.acls = acls or {}
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self):
```

Comment:
`__exit__` [takes 3 arguments](https://docs.python.org/3/reference/datamodel.html#object.__exit__) (exception type, exception info and traceback). Add `*args` here if you don't care about them, without it this will raise an exception instead of closing the file handle.

Reply:
super useful feedback - will make this change